spring-cloud-gcp-starters/spring-cloud-gcp-starter-pubsub/src/main/java/org/springframework/cloud/gcp/pubsub/autoconfig/GcpPubsubAutoConfiguration.java

```diff
@@ -48,49 +51,55 @@
 	public static final int DEFAULT_EXECUTOR_THREADS = 4;
 
 	@Bean
-	@ConditionalOnMissingBean
-	public ExecutorProvider executorProviderForPublisher() {
+	public ExecutorProvider publisherExecutorProvider() {
 		return FixedExecutorProvider.create(
 				Executors.newScheduledThreadPool(DEFAULT_EXECUTOR_THREADS));
 	}
 
 	@Bean
-	@ConditionalOnMissingBean
-	public ExecutorProvider executorProviderForSubscriber() {
+	public ExecutorProvider subscriberExecutorProvider() {
 		return FixedExecutorProvider.create(
 				Executors.newScheduledThreadPool(DEFAULT_EXECUTOR_THREADS));
 	}
 
 	@Bean
-	@ConditionalOnClass({GcpProperties.class, GoogleCredentials.class})
+	public ChannelProvider subscriberChannelProvider(GoogleCredentials credentials) {
```

Comment:
I think `ChannelProvider`s can be specified by end-user as well, therefore `@ConditionalOnMissingBean`, too.